core/src/main/scala/flatgraph/misc/Conversions.scala

```diff
@@ -3,7 +3,10 @@ package flatgraph.misc
 object Conversions {
   extension (i: Int) {
     def toShortSafely: Short = {
-      assert(0 <= i && i <= Short.MaxValue, throw new ConversionException(s"cannot safely downcast int with value=$i to short"))
+      assert(
```

Comment:
Again, I would not include such checks. But if you want to include it, then I guess `i == i.toShort.toInt` is more to the point.

Reply:
The reason for introducing this check is to provide a more specific error than an array-oob exception which is quite generic. 
The `.toShort.toInt` check is equivalent but more clear, so yes good suggestion, thanks

Reply:
>The reason for introducing this check is to provide a more specific error than an array-oob exception which is quite generic.

Yeah, but this is not an error condition that anyone is supposed to specifically catch and recover from. 

It is an exception where somebody is supposed to stare at the stack-trace in an error log, look at the line numbers, and also look at the relevant code path. 

Given that the person handling the exception already has that info, most manual throw error strings are pretty superfluous: For such exceptions, a source-code comment around the `throw` is equivalent to the static string part of an error message.